examples/cpp/Schrodinger2D.cpp

```diff
@@ -0,0 +1,110 @@
+#include "mole.h"   
+#include <iostream>   
+#include <cmath>   
+#include <iomanip>   
+#include <vector>   
+ 
+   
+using namespace arma;   
+   
+int main() {   
+  int p = 105; // Number of time steps for the simulation   
+  double Lxy = 1.0;   
+  int k = 2; // Order of accuracy   
+  int m = 50; // Grid points in x   
+  int n = 50; // Grid points in y   
+  int nx = 2; // Energy level in x   
+  int ny = 2; // Energy level in y   
+  double dx = Lxy / m; // Step in x   
+  double dy = Lxy / n; // Step in y   
+  double dt = dx; // Time step   
+   
+  // Define staggered grids   
+  vec xgrid = join_vert(vec({0}), linspace(dx / 2, Lxy - dx / 2, m), vec({Lxy}));   
+  vec ygrid = join_vert(vec({0}), linspace(dy / 2, Lxy - dy / 2, n), vec({Lxy}));   
+   
+  // Initialize 2D staggered grid   
+  mat X, Y;   
+  Utils utils;   
+  utils.meshgrid(xgrid, ygrid, X, Y);   
+   
+  // Initialize Laplacian operator with Robin BC   
+  Laplacian L(k, m, n, dx, dy);   
+  RobinBC BC(k, m, 1, n, 1, 1, 0);   
+  L = L + BC;   
+   
+  // Ensure the Laplacian is square   
+  int total_size = (m + 2) * (n + 2); // Total size for the grid including boundaries   
+  std::cout << "Laplacian L has size: " << L.n_rows << " x " << L.n_cols << std::endl;   
+   
+  // Hamiltonian definition   
+  auto H = [&](const vec &x) {   
+  return 0.5 * (L * x);  // This should produce a vector of size (m+2)*(n+2)   
+  };   
+   
+  // Initial conditions: Ensure psi_old is initialized with the correct size   
+  double A = 2 / Lxy;   
+   
+  // Initialize the wavefunction psi_old   
+  vec psi_old = vectorise(A * sin(nx * M_PI / Lxy * X) % sin(ny * M_PI / Lxy * Y)); // Adjust the sign of the wavefunction   

```

Comment:
Hi @arrowguy234 , you have bug here. You must have forgotten to include the wavenumbers. 
The corresponding Matlab code here is:

```
psi_old = A*sin(kx(nx)*X).*sin(ky(ny)*Y);
```

where `kx` and `ky` are defined as:

```
kx = @(nx) nx*pi/Lxy; % Wave vector in x
ky = @(ny) ny*pi/Lxy; % Wave vector in y
```
Please address this as soon as possible. This is time sensitive. Thank you

Reply:
Good morning, The issue has been adressed

Reply:
Hello @arrowguy234 , please post updated rigorous numerical values comparison that show that this C++ code now does reproduce the Matlab example.